packages/babel-plugin-mui-css/src/index.ts

```diff
@@ -0,0 +1,205 @@
+import * as vm from 'node:vm';
+import { existsSync, readFileSync, writeFileSync } from 'node:fs';
+import { dirname, basename, resolve as resolvePath } from 'node:path';
+import { packageDirectorySync } from 'pkg-dir';
+import * as Babel from '@babel/core';
+import * as ts from 'typescript';
+import { stylesToString } from './css/stylesToString.js';
+
+type BabelT = typeof Babel;
+
+const PLUGIN_NAME = 'mui-css';
+const CONFIG_FILENAME = 'mui-css.config.json';
+
+type State = {
+  enabled: boolean;
+  options: any;
+  config: ProjectConfig;
+  rules: { content: string }[];
+};
+
+type ProjectConfig = {
+  configPath: string;
+  data: any;
+  variablesCode: string | undefined;
+};
+
+const configsByPath = new Map<string, ProjectConfig | null>();
+
+export default function transformCSS({ types: t }: BabelT) {
+  return {
+    name: PLUGIN_NAME,
+
+    pre(file) {
+      const options = findSelf(file.opts.plugins).options;
+      const config = getConfig(file.opts.filename);
+      file.metadata.muiCSS = {
+        enabled: config !== null && !file.opts.filename.includes('cssVariables'),
+        options,
+        config,
+        rules: [],
+      } as State;
+    },
+
+    visitor: {
+      Program: {
+        exit(path, { file }) {
+          const state = file.metadata.muiCSS as State;
+          if (!state.enabled) {
+            return;
+          }
+          if (state.rules.length === 0) {
+            return;
+          }
+
+          const cssContent = state.rules.map((r) => r.content).join('\n');
+
+          const filePath = file.opts.filename;
+          const cssFilename = `${basename(filePath).split('.').slice(0, -1).join('.')}.autogenerated.css`;
+          const outputPath = resolvePath(dirname(filePath), cssFilename);
+          writeFileSync(outputPath, cssContent);
+
+          path.node.body.unshift(t.importDeclaration([], t.stringLiteral(`./${cssFilename}`)));
+        },
+      },
+
+      // const styles = css('prefix', {
+      //   class1: { ... },
+      // });
+      CallExpression(path, { file }) {
+        const state = file.metadata.muiCSS as State;
+        if (!state.enabled) {
+          return;
+        }
+        const {
+          callee: { name: calleeName },
+          arguments: args,
+        } = path.node;
+
+        if (calleeName !== 'css') {
```

Comment:
If its not too much work, can you also add the check to make sure it is the correct `css` being called that is imported from a specific path only and not any other `css` import ?